pkg/apis/sessions/session_state.go

```diff
@@ -30,9 +33,30 @@ type SessionState struct {
 	PreferredUsername string   `msgpack:"pu,omitempty"`
 }
 
+// CreatedAtNow sets a SessionState's CreatedAt to now
+func (s *SessionState) CreatedAtNow() {
+	now := NowFunc()
+	s.CreatedAt = &now
+}
+
+// SetExpiresOn sets an expiration
+func (s *SessionState) SetExpiresOn(exp time.Time) {
+	s.ExpiresOn = &exp
+}
+
+// ExpiresIn sets an expiration a certain duration from CreatedAt.
+// CreatedAt will be set to time.Now if it is unset.
+func (s *SessionState) ExpiresIn(d time.Duration) {
```

Comment:
The naming of this implied to me it would be expiring the duration after now, not duration after create, why do we want this to be expire after create rather than expire after now?

Reply:
I was torn on this implementation -- in practicality, this is consumed by OAuth redeem responses that have the `expires_in` field (analogous to the other version that had explicit `ExpiresOn`).

They immediately create a fresh session, then set the expires to the that expires in duration. I guess the time drift between these multiple calls is practically negligible, I just wanted to perfectly align the TTL difference between createdAt/expiresOn to equal the `expires_in`.

Reply:
Would `ExpiresAfter` make it any more obvious? :thinking: 

I get what you're trying to do here and don't have an issue with the logic, just wish we could come up with a more obvious name

Reply:
I was thinking making it exactly `ExpiresIn` signifies it should be used with any OAuth providers that use the `expires_in` field in their JSON response.  Where the other `SetExpiresOn` (naming conflict with `ExpiresOn` field) should be used with any OAuth providers that opt for `expires_on` in their response schema.

1:1 alignment with the OAuth schema was my goal for clarity when people implement (I agree, the name doesn't fully reflect what the logic does 😅 )